src/backend/distributed/planner/multi_explain.c

```diff
@@ -1483,7 +1483,9 @@ WrapQueryForExplainAnalyze(const char *queryString, TupleDesc tupleDesc)
 		}
 
 		Form_pg_attribute attr = &tupleDesc->attrs[columnIndex];
-		char *attrType = format_type_with_typemod(attr->atttypid, attr->atttypmod);
+		char *attrType = format_type_extended(attr->atttypid, attr->atttypmod,
```

Comment:
What about using `FORMAT_TYPE_INVALID_AS_NULL` as well, and throwing an error if it returns NULL to be on the safe side ?

```c
 * - FORMAT_TYPE_INVALID_AS_NULL
 *			if the type OID is invalid or unknown, return NULL instead of ???
 *			or such
```

Reply:
Or let's not use `FORMAT_TYPE_INVALID_AS_NULL` flag since the function itself would throw an error anyway if the type is invalid, unless we pass `FORMAT_TYPE_ALLOW_INVALID` flag. Plus `FORMAT_TYPE_INVALID_AS_NULL` is introduced in PG14.